packages/experimental-utils/src/ts-eslint/Rule.ts

```diff
@@ -209,13 +209,13 @@ interface RuleContext<
    * the root of the AST and continuing through the direct parent of the current node.
    * This array does not include the currently-traversed node itself.
    */
-  getAncestors(): TSESTree.Node[];
+  getAncestors(): readonly TSESTree.Node[];
```

Comment:
This would be a breaking change because it's not uncommon for people to do `getAncestors().reverse()`

Also - it isn't necessary because `getAncestors` returns a new array each invocation:
https://github.com/eslint/eslint/blob/d867f8100737bb82742debee2b5dc853c5f07c91/lib/linter/linter.js#L789-L797